src/loopfilter.rs

```diff
@@ -384,3 +392,702 @@ extern "C" {
+                    + q3 * 2
+                    + q4 * 2
+                    + q5
+                    + q6
+                    + q6
+                    + q6
+                    + q6
+                    + 8
+                    >> 4)
+                    .as_::<BD::Pixel>();
+                *dst.offset((strideb * 4) as isize) = (p1
+                    + p0
+                    + q0
+                    + q1
+                    + q2
+                    + q3 * 2
+                    + q4 * 2
+                    + q5 * 2
+                    + q6
+                    + q6
+                    + q6
+                    + q6
+                    + q6
+                    + 8
+                    >> 4)
+                    .as_::<BD::Pixel>();
+                *dst.offset((strideb * 5) as isize) = (p0
+                    + q0
+                    + q1
+                    + q2
+                    + q3
+                    + q4 * 2
+                    + q5 * 2
+                    + q6 * 2
+                    + q6
+                    + q6
+                    + q6
+                    + q6
+                    + q6
+                    + 8
+                    >> 4)
+                    .as_::<BD::Pixel>();
+            } else if wd >= 8 && flat8in != 0 {
+                *dst.offset(strideb * -(3 as c_int) as isize) =
+                    (p3 + p3 + p3 + 2 * p2 + p1 + p0 + q0 + 4 >> 3).as_::<BD::Pixel>();
+                *dst.offset(strideb * -(2 as c_int) as isize) =
+                    (p3 + p3 + p2 + 2 * p1 + p0 + q0 + q1 + 4 >> 3).as_::<BD::Pixel>();
+                *dst.offset(strideb * -(1 as c_int) as isize) =
+                    (p3 + p2 + p1 + 2 * p0 + q0 + q1 + q2 + 4 >> 3).as_::<BD::Pixel>();
+                *dst.offset((strideb * 0) as isize) =
+                    (p2 + p1 + p0 + 2 * q0 + q1 + q2 + q3 + 4 >> 3).as_::<BD::Pixel>();
+                *dst.offset((strideb * 1) as isize) =
+                    (p1 + p0 + q0 + 2 * q1 + q2 + q3 + q3 + 4 >> 3).as_::<BD::Pixel>();
+                *dst.offset((strideb * 2) as isize) =
+                    (p0 + q0 + q1 + 2 * q2 + q3 + q3 + q3 + 4 >> 3).as_::<BD::Pixel>();
+            } else if wd == 6 && flat8in != 0 {
+                *dst.offset(strideb * -(2 as c_int) as isize) =
+                    (p2 + 2 * p2 + 2 * p1 + 2 * p0 + q0 + 4 >> 3).as_::<BD::Pixel>();
+                *dst.offset(strideb * -(1 as c_int) as isize) =
+                    (p2 + 2 * p1 + 2 * p0 + 2 * q0 + q1 + 4 >> 3).as_::<BD::Pixel>();
+                *dst.offset((strideb * 0) as isize) =
+                    (p1 + 2 * p0 + 2 * q0 + 2 * q1 + q2 + 4 >> 3).as_::<BD::Pixel>();
+                *dst.offset((strideb * 1) as isize) =
+                    (p0 + 2 * q0 + 2 * q1 + 2 * q2 + q2 + 4 >> 3).as_::<BD::Pixel>();
+            } else {
+                let hev = ((p1 - p0).abs() > H || (q1 - q0).abs() > H) as c_int;
+
+                fn iclip_diff(v: c_int, bitdepth_min_8: u8) -> i32 {
```

Comment:
If this isn't a closure, you can add `#[inline(always)]` to guarantee inlining.

Reply:
If we add `#[inline(always)]` here, someone reading this code would be right to assume that inlining that function makes a difference performance-wise. I double checked with `perf stat` and there is no difference between never inlining and always inlining. 

I such cases, I would prefer we let the compiler chose. If someone does a build that is optimized for size (`-s` or `-z`), the compiler can choose not to inline whereas a build optimized for speed (e.g. `--release`) will probably inline this call (which indeed appears to be the case on x86).

Reply:
If that's the case, I think a closure would be better here then, as then we don't need `bitdepth_min_8` as an arg, just like it wasn't in the C `iclip_diff` macro.  But I thought you wanted to ensure inlining to match how C macros ensure inlining; in that case, an inner `fn` like this with `#[inline(always)]` will ensure that.  Overall, I'd lean in the direction of letting the compiler choose (as you said, the end user may want to optimize for different things), and so a closure would be best in that case.